BootloaderCommonPkg/Library/UefiVariableLib/UefiVariableLib.c

```diff
@@ -842,6 +842,9 @@ FindVariableEx (
     // The IndexTable->Index[Index] records the distance of two neighbouring VAR_ADDED type variables.
     //
     for (Offset = 0, Index = 0; Index < IndexTable->Length; Index++) {
+      if (Index >= UEFI_VARIABLE_INDEX_TABLE_VOLUME) {
+        return EFI_OUT_OF_RESOURCES;
+      }

```

Comment:
Can we do this before for loop? I think EFI_BAD_BUFFER_SIZE would be better.
```
if (IndexTable->Length >= UEFI_VARIABLE_INDEX_TABLE_VOLUME) {
  return EFI_BAD_BUFFER_SIZE;
}
for ()
```

Reply:
Not sure the exact KW issue reported, but the original logic should come from EDKII https://github.com/tianocore/edk2/blob/master/MdeModulePkg/Universal/Variable/Pei/Variable.c.
Maybe check with the origin code if there is any update.

Reply:
@myguitar makes sense. Wil move.

@gdong1 No changes here in the new EDK2 driver.